startup.sh

```diff
@@ -21,24 +21,22 @@ update_cron() {
   /usr/local/bin/update_cron.sh
 }
 
-# File to indicate if an update is already scheduled
-UPDATE_SCHEDULED="/tmp/update_scheduled.lock"
+# Lock file for preventing concurrent updates
+UPDATE_LOCK="/tmp/update_cron.lock"
 
-# Wrapper function to handle concurrent events
+# Wrapper function to handle concurrent events with proper locking
 handle_event() {
-  if [ -f "$UPDATE_SCHEDULED" ]; then
-    # If update is already scheduled or running, mark the need for a subsequent update
-    touch "$UPDATE_SCHEDULED"
-  else
-    # Schedule the update
-    touch "$UPDATE_SCHEDULED"
-    while [ -f "$UPDATE_SCHEDULED" ]; do
-      # Do not rush
-      sleep 2
-      rm -f "$UPDATE_SCHEDULED"
-      update_cron
-    done
-  fi
+  # Use flock to ensure only one update process runs at a time
+  (
+    flock -x 9 || {
+      log "Another update is already running, skipping this event"
```

Comment:
This failover won't be triggered unless you provide `-n` parameter because `-x` waits for the lock to release by default

But we probably don't want to have this failover in the first place, otherwise it would discard the events after the one that triggred the update so it's possible that the contab ends up in not desired state. Example with this failover in place:

- An event received for labels update
- The update of the crontab started
- Labels updated again shortly until the crontab update is completed
- The last even is discarded and you end up with crontab state not matching the labels

So I would not add `-n` but instead remove the failover `||  {...}` so it would not confuse anyone looking at it

Reply:
Thanks, you are right.